lib/nolibc/include/dirent.h

```diff
@@ -57,17 +76,16 @@ long           telldir(DIR *);
 #define DT_WHT 14
 #define IFTODT(x) ((x)>>12 & 017)
 #define DTTOIF(x) ((x)<<12)
-int getdents(int, struct dirent *, size_t);
+int getdents(int fd, struct dirent *dirp, size_t count);
+int getdents64(int fd, struct dirent64 *dirp, size_t count);
 #endif
 
 #ifdef _GNU_SOURCE
-int versionsort(const struct dirent **, const struct dirent **);
+int versionsort(const struct dirent64 **, const struct dirent64 **);
 #endif
 
 #if defined(_LARGEFILE64_SOURCE) || defined(_GNU_SOURCE)
-#define dirent64 dirent
-#define readdir64 readdir
-#define readdir64_r readdir_r
+
 #define scandir64 scandir
```

Comment:
Do these functions also need special handling in the context of `dirent`? If yes, can you provide it or at least leave a `TODO` comment? If not, can you add a comment why those do not need to be handled special?

Reply:
The `scandir64` I just missed. I provided it now.

The `alphasort` and `versionsort` are not provided by the Unikraft core, so we can just let the defines there. I added a comment about that in the file.